test/integration/auth/auth.prose.test.ts

```diff
@@ -0,0 +1,373 @@
+            .getCalls()
+            .filter(c => c.thisValue.id !== '<monitor>') // ignore all monitor connections
+            .filter(
+              c => c.args[1][process.env.MONGODB_API_VERSION ? 'hello' : LEGACY_HELLO_COMMAND]
+            );
+
+          expect(calls).to.have.length(1);
+          const handshakeDoc = calls[0].args[1];
+          expect(handshakeDoc).to.have.property('speculativeAuthenticate');
+          sinon.restore();
+          await client.close();
+        }
+      });
+    });
+
+    // todo(NODE-5621): fix the issue with unicode characters.
+    describe.skip('Step 4', function () {
+      /**
+       * Step 4
+       * To test SASLprep behavior, create two users:
+       * username: "IX", password "IX"
+       * username: "u2168" (ROMAN NUMERAL NINE), password "u2163" (ROMAN NUMERAL FOUR)
+       * To create the users, use the exact bytes for username and password without SASLprep or other normalization and specify SCRAM-SHA-256 credentials:
+       * db.runCommand(\{createUser: 'IX', pwd: 'IX', roles: ['root'], mechanisms: ['SCRAM-SHA-256']\})
+       * db.runCommand(\{createUser: 'u2168', pwd: 'u2163', roles: ['root'], mechanisms: ['SCRAM-SHA-256']\})
+       * For each user, verify that the driver can authenticate with the password in both SASLprep normalized and non-normalized forms:
+       * User "IX": use password forms "IX" and "Iu00ADX"
+       * User "u2168": use password forms "IV" and "Iu00ADV"
+       * As a URI, those have to be UTF-8 encoded and URL-escaped, e.g.:
+       * mongodb://IX:IX\@mongodb.example.com/admin
+       * mongodb://IX:I%C2%ADX\@mongodb.example.com/admin
+       * mongodb://%E2%85%A8:IV\@mongodb.example.com/admin
+       * mongodb://%E2%85%A8:I%C2%ADV\@mongodb.example.com/admin
+       */
+      let utilClient;
+      const users = [
+        {
+          username: 'IX',
+          password: 'IX',
+          mechanisms: ['SCRAM-SHA-256']
+        },
+        {
+          username: '\u2168',
+          password: '\u2163',
+          mechanisms: ['SCRAM-SHA-256']
+        }
+      ];
+
+      before(async function () {
+        utilClient = this.configuration.newClient();
+        const db = utilClient.db('admin');
+
+        const createUserCommands = users.map(user => ({
+          createUser: user.username,
+          pwd: user.password,
+          roles: ['root'],
+          mechanisms: user.mechanisms
+        }));
+
+        await Promise.all(createUserCommands.map(cmd => db.command(cmd)));
+      });
+
+      after(async function () {
+        const db = utilClient.db('admin');
+        await Promise.all(users.map(user => db.removeUser(user.username)));
+        await utilClient?.close();
+      });
+
+      [
+        { username: 'IX', password: 'IX' },
+        { username: 'IX', password: 'I\u00ADX' },
+        { username: '\u2168', password: 'IV' },
+        { username: '\u2168', password: 'I\u00ADV' }
+      ].forEach(({ username, password }) => {
```

Comment:
```suggestion
      for (const { username, password } of [
        { username: 'IX', password: 'IX' },
        { username: 'IX', password: 'I\u00ADX' },
        { username: '\u2168', password: 'IV' },
        { username: '\u2168', password: 'I\u00ADV' }
      ]) {
```

optional code modernization

Reply:
Done